src/points/ActivityCardSection.tsx

```diff
@@ -1,132 +1,122 @@
-        <ActivityCard
-          key="more-coming"
-          activityId="more-coming"
-          pointsAmount={pointsAmount}
-          onPress={onCardPress}
-        />
-      )
+  function mapActivityToCardProps(activity: PointsActivity): ActivityCardProps {
+    switch (activity.activityId) {
+      case 'create-wallet':
+        return {
+          ...activity,
+          title: t('points.activityCards.createWallet.title'),
+          icon: <Celebration />,
+        }
+      case 'swap':
+        return {
+          ...activity,
+          title: t('points.activityCards.swap.title'),
+          icon: <SwapArrows />,
+          onPress: () =>
+            onCardPress({
+              ...activity,
+              title: t('points.activityCards.swap.bottomSheet.title'),
+              body: t('points.activityCards.swap.bottomSheet.body', {
+                pointsValue: activity.pointsAmount,
+              }),
+              cta: {
+                text: t('points.activityCards.swap.bottomSheet.cta'),
+                onPress: () => {
+                  navigate(Screens.SwapScreenWithBack)
+                },
+              },
+            }),
+        }
+      default:
+        // To catch any missing cases at compile time
+        const assertNever: never = activity.activityId
+        return assertNever
     }
+  }
 
-    if (!cards.length) {
-      return <View key={pointsAmount}></View>
-    }
+  const preparedActivities = pointsActivities.map(mapActivityToCardProps).sort(sortByAmountAndTitle)
 
-    return (
-      <View
-        testID={`PointsActivitySection-${pointsAmount}`}
-        key={pointsAmount}
-        style={styles.pointsSection}
-      >
-        <View style={styles.pointsSectionHeader}>
-          <View style={styles.hr} />
-          <View style={styles.pointsSectionHeaderAmountContainer}>
-            <Text style={styles.pointsSectionHeaderAmount}>{pointsAmount}</Text>
-            <LogoHeart size={16} />
-          </View>
-          <View style={styles.hr} />
-        </View>
-        <View style={styles.pointsSectionContent}>{cards}</View>
-      </View>
-    )
-  }
+  const incompleteActivities = preparedActivities.filter(({ completed }) => !completed)
+  const completedActivities = preparedActivities.filter(({ completed }) => completed)
 
   return (
     <View style={styles.container}>
-      <View style={styles.textContainer}>
-        <Text style={styles.title}>{t('points.activitySection.title')}</Text>
-        <Text style={styles.body}>{t('points.activitySection.body')}</Text>
+      <Text style={styles.title}>{t('points.activitySection.title')}</Text>
+      <Text style={styles.body}>{t('points.activitySection.body')}</Text>
+      <View style={styles.cardsContainer}>
+        {incompleteActivities.map(renderActivityCard)}
+        <MoreComingCard />
+        {completedActivities.map(renderActivityCard)}
       </View>
-      {pointsSections.map(makeSection)}
     </View>
   )
 }
 
+function MoreComingCard() {
```

Comment:
this is fine but i wonder if it'd be easier to have this in ActivityCard so that it can share styles?

Reply:
despite looking like an ActivityCard it behaves quite differently:
* is unable to tap
* doesn't have a `PointActivityId` to track

so instead of branching to handle those specifics in the ActivityCard, I opted for this separate component

Reply:
yes totally - i just wondered if it'd be better for it to live in the ActivityCard file, so that it can share the font and spacing styles with the ActivityCard. I agree that it shouldn't be the same component

Reply:
ah, sure! great idea!